src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -588,9 +588,19 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table) {
           sym.info = readType()
           ValDef(sym.asTerm, readRhs(localCtx))
         case TYPEDEF | TYPEPARAM =>
-          if (sym.isClass)
+          def addCompanionMethod(target: Symbol, owner: Symbol) = {
```

Comment:
This method is equivalent to `registerCompanionPair` in scala2Unpickler. @VladimirNik, would you consider removing code duplication?